### ISSUE TITLE
create e2e test freamwork for scheduler

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#105329]
Hello team.

#### What would you like to be added:

create e2e test framework for scheduler on the new repo on kubernetes-sigs org (or in kubernetes/kubernetes as staging repo.)

This can be achieved by mimicking the mechanism of the scheduler simulator.

#### summary

With it, users can easily test the behavior of their custom schedulers or plugins.

In e2e framework, we can create 
- funcs to set up all components we need to simulate scheduler
  - connect to etcd
  - setup and start api-server
  - setup and start scheduler (or custom scheduler)
- funcs to get/create/edit/delete resources to simulate a cluster.
  - we can see where the pod went
- funcs to get the results of plugins
  - filtering results
  - scoring results
  - normalized results
  - score after applying plugin weights
- funcs to change the scheduler's configuration
  - change the scheduler's configuration with restarting scheduler.

What do you all think about this idea? 
I'd like to work on this if we can reach an agreement.

#### Why is this needed:

Now if users want to check the behavior of their custom schedulers or plugins, they have to set up e2e testing with real kubernetes cluster.

#### additional note:

For kubernetes/kubernetes, we can use this e2e framework for tests/integration/scheduler_perf (test to check the performance of scheduler).

https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf

In other words, this issue is "extracting the core logic of kube-scheduler-simulator in a way that makes it easy to use for testing".
Maybe, we will write a similar code as the simulator for this e2e test framework. And we can change the simulator's code to use e2e framework in the future.

/sig scheduling
/assign

**Comment by sanposhiho** [comment]
The scheduler simulator was developed based on existing integration tests framework and scheduler_perf originally. 
(So, this issue means, in other words, "move integration tests framework to another repo and improve it with reference to kube-scheduler-simulator".)

So, basically, they're similar, but kube-scheduler-simulator has features that the framework doesn't have -- visible the result of plugins by wrapping plugins internally.
Now it only supports the result of filter/score plugins, but we can support plugins for other extension points in the same way. And we also can support user-custom plugins in the same way.

And for the repo, It is more useful for users to place the framework out of kubernetes/kubernetes, because importing k8s.io/kubernetes is tricky. And we don't provide any information about the framework and I think very few people know about the framework and how to use it for e2e testing. We can provide the guides for users to describe how to use it in detail on the repo.

**Comment by sanposhiho** [comment]
Thanks for having the discussion.

/close
